### PR TITLE
[design-system]: update to 1.3.0

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* Updates the design-system dependencies
* Updates snapshots
* Updates imports to use the root (because tree shaking & module federation is now a thing)
* Use `find-root` package to find the roots of `node_modules` in the alias (see next paragraph for why)

### Why is it needed?

Originally we applied the fix of `require.resolve(${moduleName}/package.json)` which implicitly uses the node resolution methods therefore using the `exports` property of a package.json. Therefore where someone has used a conditional submodule:

```json
{
	"exports": {
		"/*": "/dist/*"
	}
}
```

the resolution will fail because the file _may not be there_ therefore I've opted to use `findRoot` where you can just pass the defacto CJS export of _any_ module and it will find the root.

### Note

If you're going to locally test it you'll want to run something like `yarn lerna clean && rm -rf node_modules && yarn setup` to clean the entire repo and start from scratch to avoid any deps caches.

### CODE REVIEW

File: packages/core/admin/webpack.alias.js [hunk 1]
```diff
@@ -40,7 +41,7 @@ const aliasExactMatch = [
 // See https://webpack.js.org/configuration/resolve/
 module.exports = {
   ...aliasExactMatch.reduce((acc, moduleName) => {
-    acc[`${moduleName}$`] = path.dirname(require.resolve(`${moduleName}/package.json`));
+    acc[`${moduleName}$`] = findRoot(require.resolve(moduleName));
```
Comment (joshuaellis) [inline packages/core/admin/webpack.alias.js:44]: This is the alias change i talk about in the PR description.

File: packages/core/admin/webpack.config.js [hunk 2]
```diff
@@ -169,7 +169,12 @@ module.exports = ({
             },
           },
         },
-
+        {
```
Comment (gu-stav) [inline packages/core/admin/webpack.config.js:172]: Can you add a comment please? In retrospective the commit message "Fix webpack" won't help much.

**Review by gu-stav — APPROVED** [review]
Works well for me ✨

**Comment by jcAtRsi** [comment]
I think something may have happened after this merge. When a plugin is importing icons or any of the design-system/v2 items (such as Pagination), build errors are generated. In the prior version (4.5.1), imports were fine. More info here: https://discord.com/channels/811989166782021633/1048343064143478794

**Pull request merged**